picodom.d.ts

```diff
@@ -0,0 +1,30 @@
+declare namespace picodom {
+  export interface VNode<D> {
+    tag: string
+    data?: D
+    children: VNode<{}>
+  }
+
+  export interface Component<D> {
+    (data?: D, ...children: VNode<{}>[]): VNode<D>
```

Comment:
Does this map to the last line in h.js: `tag(data, children)`? If so, should this be 
```typescript
export type VNodeChildren = VNode<{}> | string;

export interface Component<D> {
    (data?: D, children: VNodeChildren[]): VNode<D>
}
```

Bonus, you can reuse `VNodeChildren` in the `VNode` interface.